src/Microsoft.Diagnostics.DebugServices.Implementation/SpecialDiagInfo.cs

```diff
@@ -0,0 +1,137 @@
+        private const ulong SpecialDiagInfoAddress64 = 0x00007ffffff10000;
+        private const ulong SpecialDiagInfoAddress32 = 0x7fff1000;
+
+        [StructLayout(LayoutKind.Sequential)]
+        private struct SpecialDiagInfoHeader
+        {
+            public const int SignatureSize = 16;
+            public fixed byte Signature[SignatureSize];
+            public int Version;
+            public ulong ExceptionRecordAddress;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct EXCEPTION_RECORD64
+        {
+            public uint ExceptionCode;
+            public uint ExceptionFlags;
+            public ulong ExceptionRecord;
+            public ulong ExceptionAddress;
+            public uint NumberParameters;
+            public uint __unusedAlignment;
+            public fixed ulong ExceptionInformation[15]; //EXCEPTION_MAXIMUM_PARAMETERS
+        }
+
+        private readonly ITarget _target;
+        private readonly IMemoryService _memoryService;
+
+        public SpecialDiagInfo(ITarget target, IMemoryService memoryService)
+        {
+            _target = target;
+            _memoryService = memoryService;
+        }
+
+        private ulong SpecialDiagInfoAddress
+        {
+            get
+            {
+                if (_target.OperatingSystem == OSPlatform.OSX)
+                {
+                    if (_memoryService.PointerSize == 8)
+                    {
+                        return SpecialDiagInfoAddressMacOS64;
+                    }
+                }
+                else if (_target.OperatingSystem == OSPlatform.Linux)
+                {
+                    if (_memoryService.PointerSize == 8)
+                    {
+                        return SpecialDiagInfoAddress64;
+                    }
+                    else
+                    {
+                        return SpecialDiagInfoAddress32;
+                    }
+                }
+                return 0;
+            }
+        }
+
+        public static ICrashInfoService CreateCrashInfoService(IServiceProvider services)
+        {
+            EXCEPTION_RECORD64 exceptionRecord;
+
+            SpecialDiagInfo diagInfo = new(services.GetService<ITarget>(), services.GetService<IMemoryService>());
+            exceptionRecord = diagInfo.GetExceptionRecord();
+
+            if (exceptionRecord.ExceptionCode == CrashInfoService.STATUS_STACK_BUFFER_OVERRUN &&
+                exceptionRecord.NumberParameters >= 4 &&
+                exceptionRecord.ExceptionInformation[0] == CrashInfoService.FAST_FAIL_EXCEPTION_DOTNET_AOT)
+            {
+                uint hresult = (uint)exceptionRecord.ExceptionInformation[1];
+                ulong triageBufferAddress = exceptionRecord.ExceptionInformation[2];
+                int triageBufferSize = (int)exceptionRecord.ExceptionInformation[3];
+
+                Span<byte> buffer = stackalloc byte[triageBufferSize];
```

Comment:
You shouldn't `stackalloc` a size that came from ExceptionInformation.  This could grow way out of proportion if the dump comes in with a bad (or malicious) exception record.  ArrayPool is suitable if you are worried about the number of times this function is called, just using `new` is fine though, since I think we'd only see this called once or twice?